### DISCUSSION TITLE
[API Gateway] How to add query parameter only to CRUD requests

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#346] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11.4.6

### Description

Hi! I would like to have a default query parameter (the sort parameter `_s`) to be appended by default to all the incoming HTTP calls that will be forwarded to the CRUD Service.
 
- which is the most suited API Gateway for this task: nginx or envoy?
- which is the advanced section I have to edit?

Thanks!

**Comment by davidebianchi** [comment]
Hi @flower-of-the-bridges! 
Mmh interesting, so you need to sort all the collections in a specific way? Do you have some use case?

It is possible both with nginx and envoy, but with envoy is easier to specify specific endpoints. Unfortunately, [it seems that](https://github.com/envoyproxy/envoy/issues/2098) there is not a specific command to add the query parameter, but it is always possible to add it using the [lua filter](https://docs.mia-platform.eu/docs/development_suite/api-console/advanced-section/api-gateway-envoy/extensions#lua-scripts).

Despite this, if you need this behaviour only on crud-service what do you think about adding this as a collection configuration? It could be a default sorting parameters (or, if needed to all instance, maybe also some global configuration).

**Reply by flower-of-the-bridges** [reply]
Hi @davidebianchi 

Thanks for the references! 

In my case, many applications perform pagination over large collections: MongoDB applies natural ordering by default, which leads to possible record duplication between different calls. 

Since I would have to change the business logic of all the applications to include the application, adding a default server-side parameter to have a default sorting mechanism would be easier. Does this feature already exist on the crud service?

**Reply by davidebianchi** [reply]
No, it does not exists but I think it could be interesting to change some default parameters (it is already possible to change, for example, the default state of a record).

**Reply by davidebianchi** [reply]
I add @TheRusCompass and @danibix95 to the conversation, as they are follow the crud-service development. What do you think?

**Reply by danibix95** [reply]
Wouldn't creating a MongoDB View over the collection be sufficient?  
The view needs to be configured with the same data model of the collection, but in the aggregation pipeline it would be possible to set a `$sort` step, which would be applied to all the requests.

**Reply by flower-of-the-bridges** [reply]
HI @danibix95 

In my case, I have more than 200 collections, so creating a MongoDB view for each of them would be difficult to manage

**Reply by fredmaggiowski** [reply, marked as answer]
What about some crud configuration for a default sorter? Implementing the sorting on the api gateway means that the default would only be available to external clients, services within the namespace would not take benefit from it 🤔

**Reply by flower-of-the-bridges** [reply]
Hi all!

Thanks @fredmaggiowski and @davidebianchi  for the suggestions, It would be very nice to have this feature on the CRUD Service

I wrote a [feature request](https://github.com/mia-platform/crud-service/issues/236) based on your ideas to the crud service main repo
